doc/sphinx/source/build/project_build_stm32.rst

```diff
@@ -70,6 +70,10 @@ and generate executable file (.bin/.hex)
                .. image:: /source/build/stm32_exclude_syscalls.png
                   :width: 600
 
+            * Select 'Convert to binary file' to generate a .bin file as a post build output
+               .. image:: /source/build/stm32_convert_to_binary.png
+                  :width: 600
+
             * Program by clicking on the "Run adxxxx_iio" option seen or by performing a copy->paste option of the .hex file seen in the STM32/Debug folder
```

Comment:
I think this also can be updated making it more relevant and straight-forward.
Run builds the code and tries to program the board but with SDP-K1 (which isn't a native STM32 dev board although we use the stm32 mcu) and no available ST-Link; debug/run configurations need to be set for the programming to happen (either OpenOCD or J-Link) and a Segger J-Link probe (in case of J-Link).

We can just mention how to build code here with a screenshot: Project->right_click->build project

And flashing part is already being covered in the 'Running a project' section.